src/LanguageServer/Impl/Implementation/LineFormatter.cs

```diff
@@ -0,0 +1,643 @@
+
+            public bool IsMultilineString
+            {
+                get
+                {
+                    if (Span.Start.Line == Span.End.Line) {
+                        return false;
+                    }
+
+                    if (Kind != TokenKind.Constant || Token == Tokens.NoneToken) {
+                        return false;
+                    }
+
+                    return Token.Value is string || Token.Value is AsciiString;
+                }
+            }
+
+            public bool IsSimpleSliceToLeft
+            {
+                get
+                {
+                    if (Kind != TokenKind.Colon) {
+                        return false;
+                    }
+
+                    var a = PrevNonIgnored;
+                    var b = a?.PrevNonIgnored;
+                    var c = b?.PrevNonIgnored;
+
+                    if (a == null) {
+                        return false;
+                    }
+
+                    if (a.Is(TokenKind.LeftBracket, TokenKind.Colon)) {
+                        return true;
+                    }
+
+                    if ((!a.IsNumber && a.Kind != TokenKind.Name) || b == null) {
+                        return false;
+                    }
+
+                    if (b.Is(TokenKind.LeftBracket, TokenKind.Colon, TokenKind.Comma)) {
+                        return true;
+                    }
+
+                    if (!b.IsUnaryOp || c == null) {
+                        return false;
+                    }
+
+                    return c.Is(TokenKind.LeftBracket, TokenKind.Colon, TokenKind.Comma);
+                }
+            }
+
+            public bool IsSimpleSliceToRight
+            {
+                get
+                {
+                    if (Kind != TokenKind.Colon) {
+                        return false;
+                    }
+
+                    var a = NextNonIgnored;
+                    var b = a?.NextNonIgnored;
+                    var c = b?.NextNonIgnored;
+
+                    if (a == null) {
+                        return false;
+                    }
+
+                    if (a.Is(TokenKind.RightBracket, TokenKind.Colon, TokenKind.Comma)) {
+                        return true;
+                    }
+
+                    if (b == null) {
+                        return false;
+                    }
+
+                    if (a.IsUnaryOp) {
+                        if (c == null) {
+                            return false;
+                        }
+                        return (b.IsNumber || b.Kind == TokenKind.Name) && c.Is(TokenKind.RightBracket, TokenKind.Colon, TokenKind.Comma);
+                    }
+
+                    return (a.IsNumber || a.Kind == TokenKind.Name) && b.Is(TokenKind.RightBracket, TokenKind.Colon, TokenKind.Comma);
+                }
+            }
+
+            public TokenExt PrevNonIgnored
+            {
+                get
+                {
```

Comment:
Formatting

Reply:
The formatting settings you gave me for VS did this. Do you have an updated set of formatting rules I can use?

Reply:
We should probably update and add settings file to the PR